test/ibmq/test_ibmq_integration.py

```diff
@@ -143,3 +148,84 @@ def test_execute_two_remote(self, provider):
         job = execute([qc, qc_extra], backend, seed_transpiler=self.seed)
         results = job.result()
         self.assertIsInstance(results, Result)
+
+    def test_ensure_backend_jobs_signature(self):
+        """Test `IBMQBackend.jobs` signature is similar to `IBMQBackendService.jobs`
+
+        The signature of `IBMQBackend.jobs` is similar to the signature of
+        `IBMQBackendService.jobs` if its parameter list is a subset of the
+        parameter list of `IBMQBackendService.jobs`.
+        """
+        backend_cls = IBMQBackend
+        backend_service_cls = IBMQBackendService
+
+        # Retrieve parameter lists for both classes.
+        backend_jobs_args = getattr(
+            getfullargspec(backend_cls.jobs), 'args', [])
+        provider_backends_jobs_args = getattr(
+            getfullargspec(backend_service_cls.jobs), 'args', [])
+
+        # Ensure parameter lists not empty
+        self.assertTrue(backend_jobs_args)
+        self.assertTrue(provider_backends_jobs_args)
+
+        if not all(arg in provider_backends_jobs_args for arg in backend_jobs_args):
+            # `IBMQBackend.jobs` parameter list not a subset of `IBMQBackendService.jobs`.
+
+            # Get methods fully qualified name, includes class.
+            backend_jobs_name = getattr(
+                backend_cls.jobs, '__qualname__', str(backend_cls.jobs))
+            provider_backend_jobs_name = getattr(
+                backend_service_cls.jobs, '__qualname__', str(backend_service_cls.jobs))
+
+            differing_args = set(backend_jobs_args) - set(provider_backends_jobs_args)
+            # pylint: disable=duplicate-string-formatting-argument
+            raise Exception("`{}` does not match the parameters of `{}`. "
+                            "`{}` has the extra parameter(s): {}"
+                            .format(backend_jobs_name, provider_backend_jobs_name,
+                                    backend_jobs_name, differing_args))
+
+    def test_ibmq_managed_job_signature(self):
+        """Test `ManagedResults` and `Result` contain the same public methods.
+
+        Note:
+            Aside from ensuring that the two classes contain the same public
+            methods, it is also necessary to check that the corresponding
+            methods have the same signature.
+        """
+        result_cls = Result
+        managed_results_cls = ManagedResults
```

Comment:
Not really a big deal, but was there a reason you created new variables for these classes? Rather than doing something like `self._get_class_methods(Result)` below?